babel/messages/pofile.py

```diff
@@ -276,6 +276,7 @@ def _process_comment(self, line):
                     self.locations.append((location[:pos], lineno))
                 else:
                     self.locations.append((location, None))
+            self.locations = list(set(self.locations))
```

Comment:
Same here, this should probably also use `distinct`. Also, can you explain what's happening here? Why is `self.locations` modified every time comments are processed?

Reply:
Thank you, you're right, don't need to perform this here. When the message is created then do it there instead, once and for all. Thank you again.